File: scripts/seed-dataset.ts
import "../src/server/load-env";
import { LangfuseClient } from "@langfuse/client";
import seedDataset from "../data/seed-dataset.json";
import { env } from "../src/server/env";

async function main() {
  if (!env.langfusePublicKey || !env.langfuseSecretKey) {
    throw new Error("Langfuse credentials are required to seed a dataset.");
  }

  const langfuse = new LangfuseClient({
    publicKey: env.langfusePublicKey,
    secretKey: env.langfuseSecretKey,
    baseUrl: env.langfuseBaseUrl
  });

  try {
    await langfuse.api.datasets.create({
      name: env.datasetName,
      description: "Starter workshop dataset for the Dad IT Support Agent",
      metadata: {
        source: "repo-seed",
        workshop: "langfuse-dad-it-support"
      }
    });
  } catch (error) {
    if (!(error instanceof Error) || !/exists|duplicate/i.test(error.message)) {
      throw error;
    }
  }

  for (const item of seedDataset) {
    await langfuse.api.datasetItems.create({
      id: item.id,
      datasetName: env.datasetName,
      input: item.input,
      expectedOutput: item.expectedOutput,
      metadata: item.metadata
    });
  }

  await langfuse.flush();
  console.log(`Seeded dataset "${env.datasetName}" with ${seedDataset.length} items.`);
}

void main();

File: scripts/seed_dataset.py
from __future__ import annotations

import json

from langfuse import get_client

from backend.settings import ROOT_DIR, get_settings, is_langfuse_configured


def main() -> None:
    settings = get_settings()
    if not is_langfuse_configured(settings):
        raise RuntimeError("Langfuse credentials are required to seed a dataset.")

    langfuse = get_client()

    try:
        langfuse.create_dataset(
            name=settings.dataset_name,
            description="Starter workshop dataset for the Dad IT Support Agent",
            metadata={
                "source": "repo-seed",
                "workshop": "langfuse-dad-it-support",
            },
        )
    except Exception as exc:
        if not any(marker in str(exc).lower() for marker in ("exists", "duplicate", "already")):
            raise

    seed_path = ROOT_DIR / "data" / "seed-dataset.json"
    seed_dataset = json.loads(seed_path.read_text(encoding="utf-8"))

    for item in seed_dataset:
        langfuse.create_dataset_item(
            id=item["id"],
            dataset_name=settings.dataset_name,
            input=item["input"],
            expected_output=item["expectedOutput"],
            metadata=item["metadata"],
        )

    langfuse.flush()
    print(f'Seeded dataset "{settings.dataset_name}" with {len(seed_dataset)} items.')


if __name__ == "__main__":
    main()
